backend/substrapp/tasks/tasks.py

```diff
@@ -625,6 +626,31 @@ def _do_task(client, subtuple_directory, tuple_type, subtuple, compute_plan_id,
             client.volumes.create(name=volume_id)
         model_volume[volume_id] = {'bind': '/sandbox/local', 'mode': 'rw'}
 
+    if settings.TASK['CHAINKEYS_ENABLED']:
+        label_selector = f"compute_plan={subtuple['tag']}"
+        secrets = []
+
+        k8s_client = get_k8s_client()
+        try:
+            secret_namespace = os.getenv('K8S_SECRET_NAMESPACE', 'default')
+            secrets = k8s_client.list_namespaced_secret(secret_namespace, label_selector=label_selector)
+        except ApiException as e:
+            logging.error(f'failed to fetch namespaced secrets {secret_namespace} with selector {label_selector}')
+            raise e
```

Comment:
This should be done inside the if condition line 642 as the secrets need to be fetched at the initialisation step only.